radix-engine/src/kernel/actor.rs

```diff
@@ -72,6 +70,62 @@ pub enum Actor {
     BlueprintHook(BlueprintHookActor),
 }
 
+impl CallFrameReferences for Actor {
+    fn global_references(&self) -> Vec<GlobalAddress> {
+        let mut global_refs = Vec::new();
+
+        if let Some(blueprint_id) = self.blueprint_id() {
+            global_refs.push(blueprint_id.package_address.into());
+        }
+
+        if let Actor::Method(MethodActor { object_info, .. }) = self {
+            if let OuterObjectInfo::Some { outer_object } =
+                object_info.blueprint_info.outer_obj_info
+            {
+                global_refs.push(outer_object.clone());
+            }
+        }
+
+        global_refs
+    }
+
+    fn transient_references(&self) -> Vec<NodeId> {
+        if self.is_direct_access() {
+            vec![]
+        } else {
+            self.node_id().into_iter().collect()
+        }
+    }
+
+    fn direct_access_references(&self) -> Vec<NodeId> {
+        if self.is_direct_access() {
+            self.node_id().into_iter().collect()
+        } else {
+            vec![]
+        }
+    }
```

Comment:
We should use `InternalAddress` for transient and direct access references, as we're using `GlobalAddress` for global references?